pretraining/utils/dataset_utils.py

```diff
@@ -1350,18 +1359,26 @@ def __init__(
                     )
                 # Grab new reader if new
                 if shard != last_shard:
+                    # Close previous file
+                    if shard_file is not None:
+                        shard_file.close()
                     path = os.path.join(datapath, dataset, shard)
-                    reader = pa.ipc.open_file(path)
+                    shard_file = open(path) # Closed below
+                    reader = pa.ipc.open_file(shard_file)
                     last_shard = shard
                     shardcount += 1
-                ndocs = reader.num_record_batches
-                doc_start = (ndocs * frag) // worldsize
-                doc_end = (ndocs * frag + ndocs) // worldsize
-                # Read into temp docset_slim
-                for i in range(doc_start, doc_end):
-                    doc = reader.get_batch(i)["tokens"]  # .to_pylist()
-                    docset_slim[(dataset, shardcount, i)] = doc
-            del reader
+                if reader is not None:
```

Comment:
According to the type stubs, `open_file` can return `None`, so we must check for the `None` case to make `mypy` happy.  Here I'm just skipping this shard in case of `None`, but maybe we need to surface the error instead

Reply:
I think if `reader` had somehow been None in the old code it would have errored so should probably preserve the old behavior